src/solver/opamCudf.ml

```diff
@@ -1037,11 +1037,12 @@ let get_final_universe ~version_map univ req =
     match r with
     | Some ({Algo.Diagnostic.result = Algo.Diagnostic.Failure _; _} as r) ->
       make_conflicts ~version_map univ r
-    | Some {Algo.Diagnostic.result = Algo.Diagnostic.Success _; _}(*  -> *)
-      (* fail "inconsistent return value." *)
+    | Some {Algo.Diagnostic.result = Algo.Diagnostic.Success _; _}
     | None ->
-      (* External solver did not provide explanations, hopefully this will *)
-      check_request ~version_map univ req
+      raise (Solver_failure
+        "The current solver could not find a solution but dose3 could. \
+         This is probably a bug in the current solver. Please file a bug-report \
+         on the opam bug tracker: https://github.com/ocaml/opam/issues/")
```

Comment:
I'll double-check, but I believe this actually disables display of conflicts _always_.

Note however that I have a pending patch that calls Dose's solver first all the time, since the cost is minimal when there is a solution, and the gain can be huge when there actually is none (I first did this when working on improving the conflict messages, but it turned out to be a more general improvement).

Reply:
I'm not sure to understand. What do you mean by "disables display of conflicts *always*"?
The conflicts are being displayed correctly, I've never seen any issues here in several months of use. Is there a special case where this was correctly triggered?